[inductor] correctly setup constant in the wrapper

shunting314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97723
* __->__ #97571
* #97486
* #97405

V.graph.constants like seed_cuda_0 is not handled properly in the wrapper. Recently we move the code that initializes constants from global scope to a function. That makes assigning to seed_cuda_0 creating a new local variable rather than setup the global variable.

Add 'global var_name' lines to maintain the same behavior as before.


Test:

Run the forward graph for nvidia_deeprecommender's training run. Previous fail and now pass with the fix.

Thanks @ngimel  for report the issue with repro and @Chillee  for pointing out the root cause.



cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

shunting314 committed:
[inductor] correctly setup constant in the wrapper

[ghstack-poisoned]

ngimel reviewed (approved):
Should we have a test that makes sure generated code is standalone-runnable?

shunting314 commented:
> Should we have a test that makes sure generated code is standalone-runnable?

yep, will add one

shunting314 committed:
Update on "[inductor] correctly setup constant in the wrapper"


V.graph.constants like seed_cuda_0 is not handled properly in the wrapper. Recently we move the code that initializes constants from global scope to a function. That makes assigning to seed_cuda_0 creating a new local variable rather than setup the global variable.

Add 'global var_name' lines to maintain the same behavior as before.


Test:

Run the forward graph for nvidia_deeprecommender's training run. Previous fail and now pass with the fix.

Thanks ngimel  for report the issue with repro and Chillee  for pointing out the root cause.



[ghstack-poisoned]

shunting314 committed:
Update on "[inductor] correctly setup constant in the wrapper"


V.graph.constants like seed_cuda_0 is not handled properly in the wrapper. Recently we move the code that initializes constants from global scope to a function. That makes assigning to seed_cuda_0 creating a new local variable rather than setup the global variable.

Add 'global var_name' lines to maintain the same behavior as before.


Test:

Run the forward graph for nvidia_deeprecommender's training run. Previous fail and now pass with the fix.

Thanks ngimel  for report the issue with repro and Chillee  for pointing out the root cause.



[ghstack-poisoned]

shunting314 commented:
@pytorchbot merge
